test/parallel/test-http2-client-destroy.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+const server = h2.createServer();
+server.listen(0);
+
+server.on('listening', common.mustCall(function() {
+  const port = this.address().port;
+  const headers = { ':path': '/' };
+
+  const destroyCallbacks = [
+    client => client.destroy(),
+    client => client.socket.destroy()
```

Comment:
Minor nit: Node.js core house style is to always wrap the arguments of arrow functions in params. This won't pass linting as is.